src/sage/schemes/generic/divisor_group.py

```diff
@@ -190,6 +190,33 @@ def _element_constructor_(self, x, check=True, reduce=True):
         else:
             return Divisor_generic([(self.base_ring()(1), x)], check=False, reduce=False, parent=self)
 
+    def _coerce_map_from_(self, other):
+        r"""
+        Check if there is a coercion from ``other`` to ``self``.
+        This is used to prevent divisors on different schemes from comparing as equal to each other.
```

Comment:
```suggestion
        Return if there is a coercion map from ``other`` to ``self``.

        There is a coercion from another divisor group if the
        schemes are equal and there is a coercion map from
        the base rings.
```
As I write this, I wonder if instead we want to check instead `self.scheme().has_coerce_map_from(other.scheme())` instead of the schemes being equal. This feels like a more natural condition.

Reply:
Makes sense, yeah.